### PR TITLE
465

### PR DESCRIPTION
**Pull request by nathanlcarlson**

### CODE REVIEW

File: esg_node.py [hunk 1]
```diff
@@ -317,12 +306,13 @@ def setup_esgf_rpm_repo():
 
 def main():
     '''Main function'''
+
+    esg_setup.check_prerequisites()
+    esg_setup.init_structure()
```
Comment (William-Hill) [inline esg_node.py]: Rename init_structure() to create_esg_directories

File: base/esg_setup.py [hunk 2]
```diff
@@ -1,144 +1,108 @@
...
-    return check_os()
+    print "Checking prerequisites..."
+    check_os()
+    check_if_root()
+    check_fqdn()
+
+    return True
 
-def create_esg_directories():
+def init_structure():
```
Comment (William-Hill) [inline base/esg_setup.py]: Rename init_structure() to create_esg_directories

File: base/esg_setup.py [hunk 3]
```diff
@@ -1,144 +1,108 @@
...
-            raise UnprivilegedUserError
-        logger.debug("Root user found.")
-        return True
-    except UnprivilegedUserError:
-        logger.exception("\nMust run this program with root's effective UID\n\n")
-        esg_functions.exit_with_error(1)
+        assert assertion, err_msg
+    except AssertionError:
+        logger.error(err_msg)
+        sys.exit(1)
```
Comment (William-Hill) [inline base/esg_setup.py]: re-raise exception instead of sys.exit()

File: base/esg_setup.py [hunk 4]
```diff
@@ -1,144 +1,108 @@
...
-    except WrongOSError:
-        logger.exception("ESGF can only be installed on versions 6 of Red Hat, CentOS or Scientific Linux x86_64 systems")
-        esg_functions.exit_with_error(1)
-    else:
-        print "Operating System = {OS} {version}".format(OS=release_version[0], version=release_version[1])
-        return True
+        fqdn = socket.getfqdn()
+    except socket.error as err:
+        logger.error("%s, %s", err_msg, str(err))
+        sys.exit(1)
```
Comment (William-Hill) [inline base/esg_setup.py]: re-raise exception instead of sys.exit()

File: base/esg_setup.py [hunk 5]
```diff
@@ -1,144 +1,108 @@
...
             pybash.mkdir_p(directory)
-    os.chmod(config["esg_etc_dir"], 0777)
 
-def init_structure():
-
-    create_esg_directories()
+    os.chmod(config["esg_etc_dir"], 0777)
 
     #Create esgf.properties file
     if not os.path.isfile(config["property_file"]):
```
Comment (William-Hill) [inline base/esg_setup.py]: Remove this if statement

**Comment by nathanlcarlson** [comment]
Resolves #465

**Pull request merged**